src/components/drawer/Drawer.tsx

```diff
@@ -135,6 +151,10 @@ export type DrawerProps = {
    * Header title if this drawer has a header
    */
   title?: string
+  /**
+   * Icon to display in the header next to the title (must also include title)
+   */
+  titleIcon?: ImageSourcePropType
```

Comment:
there is a fun opportunity to strongly type the requirement that titleIcon can only be present when title is: https://react-typescript-cheatsheet.netlify.app/docs/advanced/patterns_by_usecase/#props-pass-one-only-if-the-other-is-passed

Reply:
I like it!